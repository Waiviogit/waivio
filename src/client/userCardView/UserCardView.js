import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import './UserCardView.less';
import Avatar from '../components/Avatar';

const UserCardView = ({ user }) => {
  const userName = user.name || user.account;
  return (
    <React.Fragment>
      <div className="UserCardView">
        <div className="UserCardView__content">
          <div className="UserCardView__content-row">
            <Avatar username={userName} size={80} />
            <div className="UserCardView__content--name-wrap">
              <Link
                to={`/@${userName}`}
                title={userName}
                className="UserCardView__content--user-name"
              >
                {userName}
              </Link>
            </div>
          </div>
        </div>
      </div>
    </React.Fragment>
  );
};

UserCardView.propTypes = {
  user: PropTypes.shape().isRequired,
};

export default UserCardView;
