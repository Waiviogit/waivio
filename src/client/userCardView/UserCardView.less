@import (reference) '../styles/custom.less';
@import (reference) '../styles/modules/variables.less';

.UserCardView {
  font-size: 16px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  border: 1px solid @blue-hawkesBlue;
  border-radius: 4px;

  &__content {
    width: 100%;
    position: relative;
    display: flex;
    flex-flow: column nowrap;
    min-height: 80px;
    outline: none;
    margin: 12px;

    &-row {
      display: flex;
      flex-flow: row nowrap;
      width: 100%;
    }

    &--name-wrap {
      margin-left: 5px;
      cursor: auto;
      display: flex;
      align-items: baseline;
    }

    &--user-name {
      margin-right: 10px;
      color: @black;
      font-size: medium;
      line-height: 1.1em;
      white-space: normal;
      vertical-align: middle;
      font-weight: 600;
    }
  }
}

@import 'UserCardView-nightmode';
