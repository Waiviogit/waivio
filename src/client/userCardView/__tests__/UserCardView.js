import React from 'react';
import { act } from 'react-dom/test-utils';
import { shallow } from 'enzyme';
import { IntlProvider } from 'react-intl';

import UserCardView from '../UserCardView';

describe('<UserCardView />', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      user: {
        name: 'testUser',
      },
    };

    wrapper = shallow(
      <IntlProvider locale="en">
        <UserCardView {...props} />
      </IntlProvider>,
    );
    act(() => {
      wrapper.update();
    });
  });

  it('should render without exploding', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
