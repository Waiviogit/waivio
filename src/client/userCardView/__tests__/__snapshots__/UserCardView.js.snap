// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserCardView /> should render without exploding 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IntlProvider
    locale="en"
  >
    <UserCardView
      user={
        Object {
          "name": "testUser",
        }
      }
    />
  </IntlProvider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "user": Object {
        "name": "testUser",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "user": Object {
          "name": "testUser",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): Object {
    "intl": Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    },
  },
}
`;
